packages/ui-client/src/components/GenericMenu/GenericMenu.tsx

```diff
@@ -47,7 +48,11 @@ const GenericMenu = ({ title, icon = 'menu', disabled, onAction, callbackAction,
 	const isMenuEmpty = !(sections && sections.length > 0) && !(items && items.length > 0);
 
 	if (isMenuEmpty || disabled) {
-		return <IconButton small icon={icon} disabled />;
+		if (button) {
+			return cloneElement(button, { small: true, icon, disabled, title, className });
+		}
```

Comment:
Instead of passing a `button` prop, consider passing `buttonProps`. The reason for it is that there is no way to ensure the element `button` is going to be a `<IconButton />` (e.g. `button` can be `<><IconButton /></>` and `cloneElement` is going to be pass the props to the fragment instead.

Reply:
Sharing here what we discussed. Since `button` is not a new prop and was already available in the `GenericMenu` a FIXME was added with the intention of adjusting this behavior in a separate PR, since it would be beyond the scope of this PR.